pkg/types/marshal.go

```diff
@@ -0,0 +1,56 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+var errUnmarshalling = errors.New("failed to unmarshal a value")
```

Comment:
global var.  The linter should've caught this.

Reply:
It hasn't caught this, presumably because this is an unexported error, which is OK?

I fixed a couple of other globalvar issues that the linter did notice.